tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3198,4 +3200,84 @@ public function theUserShouldHaveSpaceWithRecipient(
 		}
 		Assert::assertTrue($foundRoleInResponse, "the response does not contain the $recipientType $recipient");
 	}
+
+	/**
+	 * @When /^user "([^"]*)" downloads the space "([^"]*)" using the WebDAV API/
+	 *
+	 * @param string $user
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spaceName):void {
+		$space = $this->getSpaceByName($user, $spaceName);
+		$url = $this->featureContext->getBaseUrl() . '/archiver?id=' . $space["id"];
+		$this->featureContext->setResponse(
+			HttpRequestHelper::get(
+				$url,
+				'',
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+			)
+		);
+	}
+
+	/**
+	 * @Then the downloaded :arg1 space archive should contain these files:
+	 *
+	 * @param string $type
+	 * @param TableNode $expectedFiles
+	 *
+	 * @return void
+	 *
+	 * @throws NonExistentArchiveFileException
+	 */
+	public function theDownloadedSpaceArchiveShouldContainTheseFiles(string $type, TableNode $expectedFiles):void {
+		$dataVal = $this->featureContext->getResponse()->getBody()->getContents();
+		$placeValue = [];
+		$filePos = '';
+
+		foreach ($expectedFiles->getHash() as $file) {
+			print_r($file);
+			$place = stripos($dataVal, $file['name']);
+			echo $place . "\n";
+			$placeValue[] = $place;
+		}
+
+		print_r($placeValue, true) . "\n";
+		$min = min($placeValue);
+		echo "Min = " . $min;
+
+		foreach ($expectedFiles->getHash() as $file) {
+			$place = stripos($dataVal, $file['name']);
+			if ($place == $min) {
+				$filePos = $file['name'];
+			}
+		}
+
+		echo "\n" . $filePos . "\n";
+		$fileData = strstr($dataVal, $filePos);
+		echo $fileData . "\n";
```

Comment:
Here, the position of expected files are searched. 
The data in the response before the existence of first minimum positioned file is removed to exclude the . folder( . folder is created itself while downloading space and it cannot be extracted). 
Remaining data after excluding . folder is used to check the expected files.